lib/logitech_receiver/common.py

```diff
@@ -546,18 +551,57 @@ def __getattr__(self, k):
 """Firmware information."""
 FirmwareInfo = namedtuple("FirmwareInfo", ["kind", "name", "version", "extras"])
 
-BATTERY_APPROX = NamedInts(empty=0, critical=5, low=20, good=50, full=90)
-
-BATTERY_STATUS = NamedInts(
-    discharging=0x00,
-    recharging=0x01,
-    almost_full=0x02,
-    full=0x03,
-    slow_recharge=0x04,
-    invalid_battery=0x05,
-    thermal_error=0x06,
-)
 
-
-def BATTERY_OK(status):
-    return status not in (BATTERY_STATUS.invalid_battery, BATTERY_STATUS.thermal_error)
+@dataclass
+class Battery:
+    """Information about the current state of a battery"""
+
+    level: Optional[NamedInt | int]
```

Comment:
For Python 3.7 `Optional[Union[NamedInt, int]]` should do the trick

Reply:
done